flight.py

```diff
@@ -0,0 +1,170 @@
+    """Thrown when specified direction is invalid"""
+    pass
+
+
+class InvalidPriorityException(Exception):
+    """Thrown when specified priority is invalid"""
+    pass
+
+
+def parse_args():
+    """
+
+    Take in Command Line Arguments and give back args to program.
+
+    Returns
+    -------
+    argparse.Namespace
+        Arguments assembled by argparse.
+
+    """
+    parser = argparse.ArgumentParser(
+        description=
+        'MRRDT flight command line interface. Used to fly the drone.')
+    parser.add_argument('-d', '--debug', action="store_true")
+    parser.add_argument('routine', nargs="?", default=None, type=str)
+
+    args = parser.parse_args()
+
+    if not args.debug and not args.routine:
+        logging.error("Please specify routine.")
+        sys.exit(1)
+    return args
+
+
+def create_routes(args, controller):
+    """
+
+    Constructs all routes necessary for network flight
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments assembled by argparse.
+    controller : flight.drone.drone_controller.DroneController
+        The second parameter.
+
+    Returns
+    -------
+    list
+        Blueprints to be added to Flask app
+
+    """
+    commands = Blueprint('commands', __name__, url_prefix='/commands')
+
+    @commands.route('/', methods=["GET"])
+    def list_tasks():
+        return jsonify(controller.queue), 200
+
+    @commands.route('/', methods=["POST"])
+    def push_command():
+        data = request.get_json()
+
+        if args.debug:
+            # If in debug mode
+            if not ("command" in data and "meta" in data):
+                return "command and meta required", 400
+            command = data["command"].lower()
+            meta = data["meta"]
+            try:
+                debug_add_task(controller, command, meta)
+                return "success", 200
+            except (InvalidDirectionException,
+                    InvalidPriorityException) as err:
+                return jsonify(err), 400
+            except Exception as err:
+                logging.error("Unexpected error, killing drone")
+                debug_add_task(controller, "exit", {})
+                return "Unexpected error, killing drone " + str(err), 400
+        elif args.routine:
+            # If in production mode, starting or stopping drone is only option
+            # get push data and check if start or kill
+            # if command == "start":
+            #     start routine
+            # elif command == "kill":
+            #     force into land
+
+            return "routine " + args.routine + " in progress", 200
+
+    return [commands]
+
+
+def debug_add_task(controller, command, meta):
+    if "priority" in meta:
+        priority = get_priority(meta["priority"])
+    if "direction" in meta:
+        direction = get_direction(meta["direction"])
+
+    if command == "exit":
+        controller.add_land_task(c.Priorities.HIGH)
```

Comment:
This would be a good place to put the new ExitTask, which causes the drone controller to shut down. 
```suggestion
        controller.add_exit_task(c.Priorities.HIGH)
```